frontend/packages/console-app/src/components/create-volume-snapshot/create-volume-snapshot.tsx

```diff
@@ -0,0 +1,279 @@
+
+  const [pvcName, setPVCName] = React.useState(resourceName || '');
+  const [pvcObj, setPVCObj] = React.useState(null);
+  const [snapshotName, setSnapshotName] = React.useState<string>(`${pvcName}-snapshot`);
+  const [snapshotClassName, setSnapshotClassName] = React.useState<string>();
+  const [error, setError] = React.useState('');
+  const [inProgress, setInProgress] = React.useState(false);
+  const [pvcStatus, setPVCStatus] = React.useState<string>(null);
+  const [pvcProvisioner, setPVCProvisioner] = React.useState<string>(null);
+
+  React.useEffect(() => {
+    const setValue = async () => {
+      try {
+        const pvc = await fetchK8s<K8sResourceKind>(PersistentVolumeClaimModel, pvcName, namespace);
+        const sc = await fetchK8s<StorageClassResourceKind>(
+          StorageClassModel,
+          pvc?.spec?.storageClassName,
+        );
+        setPVCObj(pvc);
+        setPVCStatus(pvc?.metadata?.deletionTimestamp ? 'Terminating' : pvc?.status?.phase);
+        setPVCProvisioner(sc?.provisioner);
+      } catch (err) {
+        setError(err);
+      }
+    };
+
+    setValue();
+  }, [pvcName, namespace]);
+
+  const handleSnapshotName: React.ReactEventHandler<HTMLInputElement> = (event) => {
+    setSnapshotName(event.currentTarget.value);
+  };
+
+  const create = (event: React.FormEvent<EventTarget>) => {
+    event.preventDefault();
+    setInProgress(true);
+    const snapshotTemplate: K8sResourceKind = {
+      apiVersion: `${VolumeSnapshotModel.apiGroup}/${VolumeSnapshotModel.apiVersion}`,
+      kind: VolumeSnapshotModel.kind,
+      metadata: {
+        name: snapshotName,
+        namespace,
+      },
+      spec: {
+        volumeSnapshotClassName: snapshotClassName,
+        source: {
+          persistentVolumeClaimName: pvcName,
+        },
+      },
+    };
+    k8sCreate(VolumeSnapshotModel, snapshotTemplate)
+      .then((resource) => {
+        setInProgress(false);
+        history.push(resourceObjPath(resource, referenceFor(resource)));
+      })
+      .catch((err) => {
+        setError(err.message || 'Could not create volume snapshot');
+        setInProgress(false);
+      });
+  };
+
+  const title = 'Create VolumeSnapshot';
+  return (
+    <div className="co-m-pane__body co-volume-snapshot__body">
+      <Helmet>
+        <title>{title}</title>
+      </Helmet>
+      <form className="co-m-pane__body-group co-m-pane__form" onSubmit={create}>
+        <h1 className="co-m-pane__heading">{title}</h1>
+        {kindObj.kind === VolumeSnapshotModel.kind && (
+          <PVCDropdown
+            namespace={namespace}
+            onChange={setPVCName}
+            id="claimName"
+            selectedKey={pvcName}
+          />
+        )}
+        {kindObj.kind === PersistentVolumeClaimModel.kind && (
+          <p>
+            Creating snapshot for claim <strong>{resourceName}</strong>
+          </p>
+        )}
+        {pvcStatus && pvcStatus !== 'Bound' && (
```

Comment:
I think you can directly use `pvcStatus !== 'Bound'` here once you define the `pvsStatus` with empty string